smartsim/_core/entrypoints/dragon.py

```diff
@@ -146,6 +185,8 @@ def main(args: argparse.Namespace) -> int:
             dragon_head_address += ":5555"
 
         zmq_authenticator = dragonSockets.get_authenticator(zmq_context)
+        zmq_authenticator.thread.authenticator.zap_socket.setsockopt(zmq.SNDTIMEO, -1)
```

Comment:
Should we pass timeouts to `get_authenticator` to allow that setup to be encapsulated there instead?

Reply:
Done